packages/embedder/ai_models/OpenAIGeneration.ts

```diff
@@ -0,0 +1,94 @@
+import OpenAI from 'openai'
+import {
+  AbstractGenerationModel,
+  GenerationModelConfig,
+  GenerationModelParams,
+  GenerationOptions
+} from './AbstractModel'
+
+const MAX_REQUEST_TIME_S = 3 * 60
+
+export type ModelId = 'gpt-3.5-turbo-0125' | 'gpt-4-turbo-preview'
+
+type OpenAIGenerationOptions = Omit<GenerationOptions, 'topK'>
+
+const modelIdDefinitions: Record<ModelId, GenerationModelParams> = {
+  'gpt-3.5-turbo-0125': {
+    maxInputTokens: 4096
+  },
+  'gpt-4-turbo-preview': {
+    maxInputTokens: 128000
+  }
+}
+
+function isValidModelId(object: any): object is ModelId {
+  return Object.keys(modelIdDefinitions).includes(object)
+}
+
+export class OpenAIGeneration extends AbstractGenerationModel {
+  private openAIApi: OpenAI | null
+  private modelId: ModelId
+
+  constructor(config: GenerationModelConfig) {
+    super(config)
+    if (!process.env.OPEN_AI_API_KEY) {
+      this.openAIApi = null
+      return
+    }
+    this.openAIApi = new OpenAI({
+      apiKey: process.env.OPEN_AI_API_KEY,
+      organization: process.env.OPEN_AI_ORG_ID
+    })
+  }
+
+  async summarize(content: string, options: OpenAIGenerationOptions) {
+    if (!this.openAIApi) {
+      const eMsg = 'OpenAI is not configured'
+      console.log('OpenAIGenerationSummarizer.summarize(): ', eMsg)
+      throw new Error(eMsg)
+    }
+    const {maxNewTokens: max_tokens = 512, seed, stop, temperature = 0.8, topP: top_p} = options
+    const prompt = `Create a brief, one-paragraph summary of the following: ${content}`
```

Comment:
Observing the differences between Zephyr and gpt-3.5-turbo, I have a feeling we may need to play with this prompt a bit to maximize which features are included in the summary. There are choices we may want to consider:
   * Preserving names vs. using pronouns if they are known
   * Attempting to summarize each thread into a chunk of text, rather than summarizing the entire block

I'm not sure yet, though, so I want to forge ahead and try and do some reporting before iterating on this pipeline.